backend/hitas/tests/apis/test_api_indices.py

```diff
@@ -0,0 +1,222 @@
+    ConstructionPriceIndexPre2005Factory,
+    MarketPriceIndexFactory,
+    MarketPriceIndexPre2005Factory,
+    MaxPriceIndexFactory,
+)
+
+_code_parameters = (
+    "url_basename,model,factory",
+    [
+        ("max-price-index", None, MaxPriceIndexFactory),
+        ("market-price-index", None, MarketPriceIndexFactory),
+        ("market-price-index-pre-2005", None, MarketPriceIndexPre2005Factory),
+        ("construction-price-index", None, ConstructionPriceIndexFactory),
+        ("construction-price-index-pre-2005", None, ConstructionPriceIndexPre2005Factory),
+    ],
+)
+
+indices = [
+    "max-price-index",
+    "market-price-index",
+    "market-price-index-pre-2005",
+    "construction-price-index",
+    "construction-price-index-pre-2005",
+]
+
+factories = [
+    MaxPriceIndexFactory,
+    MarketPriceIndexFactory,
+    MarketPriceIndexPre2005Factory,
+    ConstructionPriceIndexFactory,
+    ConstructionPriceIndexPre2005Factory,
+]
+
+# List tests
+
+
+@pytest.mark.parametrize("index", indices)
+@pytest.mark.django_db
+def test__api__indices__list__empty(api_client: HitasAPIClient, index):
+    response = api_client.get(reverse(f"hitas:{index}-list"))
+    assert response.status_code == status.HTTP_200_OK, response.json()
+    assert response.json() == {
+        "contents": [],
+        "page": {
+            "size": 0,
+            "current_page": 1,
+            "total_items": 0,
+            "total_pages": 1,
+            "links": {
+                "next": None,
+                "previous": None,
+            },
+        },
+    }
+
+
+@pytest.mark.parametrize("index,factory", zip(indices, factories))
+@pytest.mark.django_db
+def test__api__indices__list(api_client: HitasAPIClient, index, factory):
+    factory.create(month=datetime.date(2022, 1, 1), value=127.48)
+    factory.create(month=datetime.date(1999, 12, 1), value=256)
+
+    response = api_client.get(reverse(f"hitas:{index:}-list"))
+    assert response.status_code == status.HTTP_200_OK, response.json()
+    assert response.json() == {
+        "contents": [
+            {"month": "1999-12", "value": 256},
+            {"month": "2022-01", "value": 127.48},
+        ],
+        "page": {
+            "size": 2,
+            "current_page": 1,
+            "total_items": 2,
+            "total_pages": 1,
+            "links": {
+                "next": None,
+                "previous": None,
+            },
+        },
+    }
+
+
+# Create
+
+
+@pytest.mark.parametrize("index", indices)
+@pytest.mark.django_db
+def test__api__indices__create(api_client: HitasAPIClient, index):
+    response = api_client.post(reverse(f"hitas:{index}-list"), data={}, openapi_validate=False)
+    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED, response.json()
+    assert response.json() == {
+        "error": "method_not_allowed",
+        "message": "Method not allowed",
+        "reason": "Method Not Allowed",
+        "status": 405,
+    }
```

Comment:
Since creating is now allowed, how are the indices created? 🤔

Reply:
it's possible to fetch all the months that have values with `GET /indices/{index}`.
It's possible to query a specific month with `GET `/indices/{index}/month`. It will return a value or `null` if not set.
It's possible to update a specific month with `PUT `/indices/{index}/month` and set `value` to any value or `null`.

The idea here is that all the months "exist". Values for them might or might not.

Reply:
All righty, makes sense 😄